flink-end-to-end-tests/flink-sql-gateway-test/src/test/java/org/apache/flink/table/gateway/SqlGatewayE2ECase.java

```diff
@@ -126,6 +135,339 @@ public void testSqlClientExecuteStatement() throws Exception {
+                        try {
+                            List<RowData> result =
+                                    gatewayRestClient.executeStatementWithResult(
+                                            "select * from my_materialized_table_in_continuous_mode order by ds, user");
+                            return result.size() == 2;
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    },
+                    continuousWaitTime,
+                    continuousWaitPause,
+                    "Failed to wait for the result");
+
+            File savepointFolder = FOLDER.newFolder("savepoint");
+            // configure savepoint path
+            gatewayRestClient.executeStatementWithResult(
+                    String.format(
+                            "set 'execution.checkpointing.savepoint-dir'='file://%s'",
+                            savepointFolder.getAbsolutePath()));
+
+            // suspend the materialized table
+            gatewayRestClient.executeStatementWithResult(
+                    "ALTER MATERIALIZED TABLE my_materialized_table_in_continuous_mode SUSPEND");
+
+            // resume the materialized table
+            gatewayRestClient.executeStatementWithResult(
+                    "ALTER MATERIALIZED TABLE my_materialized_table_in_continuous_mode RESUME");
+
+            // send more data to the source
+            gatewayRestClient.executeStatementWithResult(
+                    "insert into streaming_source select TO_TIMESTAMP('2024-06-20 00:00:00'), 'Charlie', 'WARN'");
+
+            // verify the result
+            CommonTestUtils.waitUtil(
+                    () -> {
+                        try {
+                            List<RowData> result =
+                                    gatewayRestClient.executeStatementWithResult(
+                                            "select * from my_materialized_table_in_continuous_mode");
+                            return result.size() == 3;
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    },
+                    continuousWaitTime,
+                    continuousWaitPause,
+                    "Failed to wait for the result");
+        }
+    }
+
+    @Test
+    public void testMaterializedTableInFullMode() throws Exception {
+        Duration fullModeWaitTime = Duration.ofMinutes(5);
+        Duration fullModeWaitPause = Duration.ofSeconds(10);
+
+        // init session
+        try (GatewayController gateway = flinkResource.startSqlGateway();
+                ClusterController ignore = flinkResource.startCluster(4)) {
+
+            File catalogStoreFolder = FOLDER.newFolder("catalog-store");
+            Map<String, String> sessionProperties = new HashMap<>();
+            sessionProperties.put("table.catalog-store.kind", "file");
```

Comment:
The should be configured in Flink config.yaml file? The refresh batch job need to depend on file catalog store to get find materialized table when scheduler open a new session to trigger refresh operation.

Reply:
Currently, we have saved the initConfig of the session to the workflow info, which can be used by the scheduler for initialization during refresh.